gravitee-apim-console-webui/src/management/api/portal/subscriptions/subscriptions.component.ts

```diff
@@ -260,6 +269,45 @@ const ApiSubscriptionsComponent: ng.IComponentOptions = {
         this.query.api_key
       );
     }
+
+    doSubscribe(application: any, plan: any, customApiKey: string) {
+      this.ApiService.subscribe(this.api.id, application.id, plan.id, customApiKey).then((response) => {
+        const subscription = response.data;
+        this.NotificationService.show('A new subscription has been created.');
+        this.$state.go('management.apis.detail.portal.subscriptions.subscription', { subscriptionId: subscription.id }, { reload: true });
+      });
+    }
+
+    selectKeyMode() {
+      const dialog = {
+        controller: 'ApplicationKeyDialogController',
+        controllerAs: '$ctrl',
+        template: require('../../../application/details/subscribe/application-key.dialog.html'),
+        clickOutsideToClose: true,
+      };
+
+      return this.$mdDialog.show(dialog);
+    }
+
+    $shouldPromptForKeyMode(application: any, plan: any): IPromise<boolean> {
+      if (plan.security === PlanSecurityType.API_KEY && this.allowsSharedApiKeys && application.api_key_mode === ApiKeyMode.UNSPECIFIED) {
+        return this.$getApiKeySubscriptionsCount(application).then((response) => {
+          return response >= 1;
+        });
+      }
+      return Promise.resolve(false);
+    }
+
+    $getApiKeySubscriptionsCount(application: any): IPromise<number> {
+      return this.ApplicationService.listSubscriptions(application.id, '?expand=security').then((response) => {
+        let applicationSubscriptions = response.data as PagedResult;
```

Comment:
```suggestion
        const applicationSubscriptions = response.data as PagedResult;
```
(This make eslint unhappy)